### DISCUSSION TITLE
How to create a new ebsd after merge grains？

### DISCUSSION THREAD

**Discussion by PengWei97** [mtex-toolbox/mtex#1657] [Ask Anything]
Dear MTEX experts,

Recently, I merged the twins according to the calculation examples provided on the [Merging Grains](https://mtex-toolbox.github.io/GrainMerge.html), and got the IPF map, but I want to get **the new ebsd data** based on the `mergedGrains` without twins, and finally output it as a ctf file. How do I do it?

![image](https://user-images.githubusercontent.com/56183775/202664500-5ab49f4e-c0d1-44a3-a27a-4c72ae74518f.png)

Any suggestions or recommendations to fix the problem would be greatly appreciated.

Thank you
Wei

**Comment by kilir** [comment]
Hi,
merging grains does not change your EBSD orientation data.
Cheers,
Rüdiger

**Reply by PengWei97** [reply]
Yeah, you are right! So now I want to build a new ebsd data based on the current `mergedGrains`, how should I do it? 

In other words, how to modify the orientation of the merged twins in the original EBSD so that it becomes the orientation of its matrix grain?

Cheers,
Wei Peng

**Reply by ralfHielscher** [reply]
Would it be sufficient to set the ebsd orientations to the merged grain meanorientations? This can be done.

Ralf.

**Reply by PengWei97** [reply]
Yeah, this solution is feasible. 

Specifically, construct the correspondence between each `merge_id` and the original `grain_id` according to the output `parentId`, and then execute `ebsd.orientation = mergedGrain.orientation`. Finally, considering the `ebsd.rotation` used in `export_ang`/`export_ctf`, modify these two functions and set it to output ebsd.orientation.

![image](https://user-images.githubusercontent.com/56183775/203573000-26e50694-2dae-4d2f-a705-a82f3a0efa8a.png)

The above is my solution. By the way, are there any other hidden dangers in such the solution?

wei
